hotel/models.py

```diff
@@ -113,10 +120,18 @@ def legal_last_name(self):
         Hotel exports need split legal names, but we don't collect split legal names, so we're going to have to guess.
 
         Returns one of the following:
+            The second part of the legal name, if the legal name starts with the legal first name
             The second part of the legal name after a space, if the legal name has multiple parts
             The last name, if there is no legal name or if the legal name is just one word
         """
-        return self.legal_name.split(' ', 1)[1] if self.legal_name and ' ' in self.legal_name else self.last_name
+        if self.legal_name:
+            legal_name = self.legal_name.strip()
+            legal_first_name = self.legal_first_name.strip()
+            if legal_name.lower().startswith(legal_first_name.lower()):
```

Comment:
This will be true anytime `legal_first_name` was derived from `legal_name`, which as you point out should always be true. I suppose it's a sanity check? I don't want to assume the inner workings of another function (even if it's defined just a few lines above!)

Consider the case where this class is subclassed, and the subclass overrides the implementation of `legal_first_name`. Then this conditional would fail, and we'd want to fall back to the old method of splitting the `legal_name`.

I guess it's just paranoid defensive programming on my part! 😅